### PR TITLE
[OPS-256] Remote container build

### PR DESCRIPTION
**Pull request by alourie**
# Description

An update to the `pipeline-java-ewb` container build flow to allow remote execution and payload parsing.

# Associated tasks
- https://github.com/zepben/ci-utils/pull/19
- https://github.com/zepben/.github/pull/52

# Checklist

If any of these are not applicable, strikethrough the line `~like this~`. **Do not delete it!**. Let the reviewer decide if you should have done it.

### Code
- [x] I have performed a self review of my own code (including checking issues raised when creating the PR).
~- [ ] I have added/updated unit tests for these changes, and if not I have explained why they are not necessary.~
- [x] I have commented my code in any hard-to-understand or hacky areas.
- [x] I have handled all new warnings generated by the compiler or IDE.
- [x] I have rebased onto the target branch (usually main).

### Documentation
~- [] I have updated the changelog.~
~- [] I have updated any documentation required for these changes.~

### CODE REVIEW

**Comment by ryanjcoleman** [comment]
Task linked: [OPS-256 Update superpom java container build](https://app.clickup.com/t/6929263/OPS-256)

**Comment by alourie** [comment]
DO NOT MERGE until the `ci-utils` PR is merged.

File: .github/workflows/build-container.yml [hunk 1]
```diff
@@ -57,8 +57,28 @@ jobs:
...
+          # Handle the resulting container labels if provided
+          if [ "z${{ github.event.client_payload.labels }}" != "z" ]; then
+            # the labels come in as "label1 label2" but we need to provide them as
+            # '["label1", "label2"]'
+            # So we replace the original values with quoted ones and then wrap in braces
+
+            # Split the passed values into an array
+            array=(${{ github.event.client_payload.labels }})
+            # Print every array entry encased by double quotes
+            tags=$(printf "\"%s\"," "${array[@]}")
```
Comment (charlta) [inline .github/workflows/build-container.yml:75]: question: Does the trailing comma here cause issues? My understanding is it shouldn't be there (NOT a proper JSON array), but maybe we are lucky and it works for now?

Comment (alourie) [reply .github/workflows/build-container.yml:75]: Yes, I specifically tested it. It's used in `packer build`, so only need to be a list, so this:
```
packer build -var tags='["label1", "label2",]' ...
```
seems to be a legit run.

Comment (alourie) [reply .github/workflows/build-container.yml:75]: image

**Review by zarakay — APPROVED** [review]

File: .github/workflows/build-container.yml (hunk 1 above)
Comment (charlta) [reply .github/workflows/build-container.yml:75]: I'm guessing this is due to changes in the HCL2 format. I'm pretty sure the trailing comma would have caused issues in the old JSON format.

Comment (alourie) [reply .github/workflows/build-container.yml:75]: Yea potentially

**Pull request merged**